src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -907,6 +907,76 @@ public override IOperation VisitWhileLoop(IWhileLoopOperation operation, int? ca
             return null;
         }
 
+        public override IOperation VisitVariableDeclarationGroup(IVariableDeclarationGroupOperation operation, int? captureIdForResult)
+        {
+            // We erase variable declarations from the control flow graph, as variable lifetime information is contained in a parallel data structure.
+            foreach (var declaration in operation.Declarations)
+            {
+                HandleVariableDeclaration(declaration, captureIdForResult);
+            }
+
+            return null;
+        }
+
+        private void HandleVariableDeclaration(IVariableDeclarationOperation operation, int? captureIdForResult)
+        {
+            foreach (IVariableDeclaratorOperation declarator in operation.Declarators)
+            {
+                ILocalSymbol localSymbol = declarator.Symbol;
+
+                IOperation initializer = null;
+                SyntaxNode assignmentSyntax = null;
+                if (declarator.Initializer != null)
+                {
+                    initializer = Visit(declarator.Initializer.Value, captureIdForResult);
+                    assignmentSyntax = declarator.Syntax;
+                }
+
+                if (operation.Initializer != null)
+                {
+                    IOperation operationInitializer = Visit(operation.Initializer.Value, captureIdForResult);
+                    assignmentSyntax = operation.Syntax;
+                    if (initializer != null)
+                    {
+                        initializer = new InvalidOperation(ImmutableArray.Create(initializer, operationInitializer), semanticModel: null, operation.Syntax, type: localSymbol.Type, constantValue: default, isImplicit: true);
+                    }
+                    else
+                    {
+                        initializer = operationInitializer;
+                    }
+                }
+
+                if (initializer != null)
+                {
+                    // We can't use the IdentifierToken as the syntax for the local reference, so we use the entire declarator as the node
+                    var localRef = new LocalReferenceExpression(localSymbol, isDeclaration: true, semanticModel: null, declarator.Syntax, localSymbol.Type, constantValue: default, isImplicit: true);
+                    // prototype(dataflow): We'd like to remove ExpressionStatements from the CFG altogether, as they're useless when all you need to do is look to see if the parent is null
```

Comment:
>prototype [](start = 23, length = 9)

"prototype" should be in all caps

Reply:
Fixed all instances of this.

---
In reply to: [167021108](https://github.com/dotnet/roslyn/pull/24692#discussion_r167021108) [](ancestors = 167021108)